tests/test_compare_with_default_qubit.py

```diff
@@ -55,110 +54,83 @@ def setUp(self):
-                            raise IgnoreOperationException(
-                                'Skipping in automatic test because' + observable + 'is not valid in ProjectQ.')
-                        else:
-                            observable_pars = {}
-
-                        # apply to the first wires
-                        operation_wires = list(range(operation_class.num_wires)) if operation_class.num_wires > 1 else 0
-                        observable_wires = list(range(observable_class.num_wires)) if observable_class.num_wires > 1 else 0
-
-                        if str(operation) == "QubitStateVector":
-                            operation_wires = range(self.num_subsystems)
-
-                        operation_class(*operation_pars, wires=operation_wires)
-                        return qml.expval(observable_class(*observable_pars, wires=observable_wires))
-
-                    try:
-                        output = circuit()
-                    except IgnoreOperationException:
-                        continue
-
-                    if (operation, observable) not in outputs:
-                        outputs[(operation, observable)] = {}
-
-                    outputs[(operation, observable)][str(type(dev).__name__)+"(shots="+str(dev.shots)+")"] = output
-
-        #if we could run the circuit on more than one device assert that both should have given the same output
-        for (key,val) in outputs.items():
-            if len(val) >= 2:
-                self.assertAllElementsAlmostEqual(val.values(), delta=self.tol, msg="Outputs "+str(list(val.values()))+" of devices ["+', '.join(list(val.keys()))+"] do not agree for a circuit consisting of a "+str(key[0])+" Operation followed by a "+str(key[1])+" Expectation." )
-
-
-if __name__ == '__main__':
-    log.info('Testing PennyLane ProjectQ Plugin version ' + qml.version() + ', Device class.')
-    # run the tests in this file
-    suite = unittest.TestSuite()
-    for t in (CompareWithDefaultQubitTest, ):
-        ttt = unittest.TestLoader().loadTestsFromTestCase(t)
-        suite.addTests(ttt)
-
-    unittest.TextTestRunner().run(suite)
+                        """4-qubit circuit with layers of randomly selected gates and random connections for
+                        multi-qubit gates."""
+                        qml.BasisState(np.array([1, 0, 0, 0]), wires=[0, 1, 2, 3])
+                        np.random.seed(1967)
+                        for gates in gates_per_layers:
+                            for gate in gates:
+                                if gate.name in dev.operations:
+                                    qml.apply(gate)
+                        return qml.expval(obs)
+
+                    qnode_default = qml.QNode(circuit, default_qubit)
+                    qnode = qml.QNode(circuit, dev)
+
+                    assert np.allclose(qnode(), qnode_default(), atol=1e-3)
+
+    def test_projectq_ops(self):
+
+        results = [-1.0, -1.0]
```

Comment:
Are the results here always the same for all the circuits generated below?

Reply:
Yes always the same circuit results because it is seeded.

Reply:
Ah yes, that's right!